http-server-tck/src/main/java/io/micronaut/http/server/tck/tests/ErrorHandlerFluxTest.java

```diff
@@ -0,0 +1,149 @@
+                .body("Your request is erroneous: Cannot process request.")
+                .build()));
+    }
+
+    @Test
+    void testErrorHandlerWithFluxChunkedSignaledImmediateError() throws IOException {
+        asserts(SPEC_NAME,
+            HttpRequest.GET("/errors/flux-chunked-immediate-error"),
+            (server, request) -> AssertionUtils.assertThrows(server, request, HttpResponseAssertion.builder()
+                .status(HttpStatus.INTERNAL_SERVER_ERROR)
+                .body("Internal Server Error: Cannot process request.")
+                .build()));
+    }
+
+    @Test
+    void testErrorHandlerWithFluxChunkedSignaledDelayedError() throws IOException {
+        asserts(SPEC_NAME,
+            HttpRequest.GET("/errors/flux-chunked-delayed-error"),
+            (server, request) -> {
+                Executable e = () -> server.exchange(request);
+                Assertions.assertThrows(HttpClientException.class, e);
+            });
+    }
+
+    @Test
+    void testErrorHandlerWithFluxSingleResultSignaledError() throws IOException {
+        asserts(SPEC_NAME,
+            HttpRequest.GET("/errors/flux-single-error"),
+            (server, request) -> AssertionUtils.assertThrows(server, request, HttpResponseAssertion.builder()
+                .status(HttpStatus.BAD_REQUEST)
+                .body("Your request is erroneous: Cannot process request.")
+                .build()));
+    }
+
+    @Requires(property = "spec.name", value = SPEC_NAME)
+    @Controller("/errors")
+    static class ErrorController {
+
+        @Get("/flux-exception")
+        Flux<String> fluxException() {
+            throw new MyTestException("Cannot process request.");
+        }
+
+        @Get("/flux-single-exception")
+        @SingleResult
+        Flux<String> fluxSingleException() {
+            throw new MyTestException("Cannot process request.");
+        }
+
+        @Get("/flux-single-error")
+        @SingleResult
+        Flux<String> fluxSingleError() {
+            return Flux.error(new MyTestException("Cannot process request."));
+        }
+
+        @Get("/flux-chunked-immediate-error")
+        Flux<String> fluxChunkedImmediateError() {
+            return Flux.error(new MyTestException("Cannot process request."));
+        }
+
+        @Get("/flux-chunked-delayed-error")
+        Flux<String> fluxChunkedDelayedError() {
+            return Flux.just("1", "2", "3").handle((data, sink) -> {
+                if (data.equals("3")) {
+                    sink.error(new MyTestException("Cannot process request."));
+                } else {
+                    sink.next(data);
+                }
+            });
+        }
+
+        @Error(global = true)
+        public HttpResponse<String> handleMyTestException(HttpRequest<?> request, MyTestException exception) {
+            var error = "Your request is erroneous: " + exception.getMessage();
+            return HttpResponse.<String>status(HttpStatus.BAD_REQUEST, "Bad request")
```

Comment:
Respond a weird status code .e.g I am a teapot so we know in the test assertion we are being handled by the test error handler.